File: 1cdict/translate-with-dict.py
import re
import argparse


filedict = 'dict-test.txt'
dict_ru_en = list()

with open(filedict, "r", encoding="utf-8") as f:
    for str in f.readlines():
        params = re.findall("^(\w+) - (\w+)$", str)
        dict_ru_en.append(params)

parser = argparse.ArgumentParser()
parser.add_argument("--f", help="Входящий файл")
parser.add_argument("--o", help="Исходящий файл. Если не указать, то будет запись во входящий файл.")
args = parser.parse_args()


if (args.f):
    filename = args.f

    if (args.o):
        filename_en = args.o
    else:
        filename_en = filename
else:
    filename = 'Module.bsl'
    filename_en = 'Module-en.bsl'

lines = list()
with open(filename, "r", encoding="utf-8") as f:
    lines = f.readlines()

with open(filename_en, "w", encoding="utf-8") as f:
    for reg in dict_ru_en:
        for line in lines:
        # old_data = old_data.replace(reg[0][0], reg[0][1])
            pattern1 = '[\s(=]'+reg[0][0]+'[\s);,=]'
            pattern2 = '^'+reg[0][0]+'[\s);,=]'
            pattern3 = '[\s(=]'+reg[0][0]+'$'
            repl = reg[0][1]
            line = re.sub(pattern1, repl, line)
            line = re.sub(pattern2, repl, line)
            line = re.sub(pattern3, repl, line)
            f.write(line)
# with open(filename_en, "w", encoding="utf-8") as f:
#         f.write(old_data)

print ("Done!")

File: 1cdict/translate-with-dict.pl
use strict;
use warnings;
use utf8;
use Data::Dumper::Simple;
use 5.010;

#get parameters
my $filename;
my $filename_en;
if (scalar($ARGV)) {
    $filename = $ARGV[0];
    if ($ARGV[1]) {
        $filename_en = $ARGV[1];
    }
    else {
        $filename_en = $filename;
    }
}
else {
    $filename = 'Module.bsl';
    $filename_en = 'Module-en.bsl';
}



my $filedict = "dict.txt";
open(my $fhdict, '<:encoding(UTF-8)', $filedict)
    or die "Could not open file '$filedict' $!";

my @dict;
while (my $line = <$fhdict>) {
    if($line =~ /^#/){
        next;
    }
    $line =~ /(^\w+) - (\w+)$/;

    my %rec = ();
    $rec{'ru'} = $1;
    $rec{'en'} = $2;

    push(@dict, \%rec);
}
close($fhdict);

#warn Dumper @dict;

open(my $fh, '<:encoding(UTF-8)', $filename)
    or die "Could not open file '$filename' $!";

my @lines;
while (my $line = <$fh>) {
    push(@lines, $line);
}
close($fh);

my $doreplases = 0;
foreach my $reg (@dict) {
    foreach my $line (@lines) {
        unless ($line =~ /^\//) {
            my $old_line = $line;

	    my $ru = $reg->{ru};
	    my $en = $reg->{en};

	    $line =~ s/(?<![А-я])$ru(?![А-я])/$en/g;

            if ($line ne $old_line) {
#                print "$old_line -> $line";
                $doreplases++;
            }
        }
    }
}

open(my $fw, '>:encoding(UTF-8)', $filename_en)
    or die "Could not open file '$filename' $!";

foreach my $line (@lines) {
    print $fw $line;
}
close($fw);

print "Done $doreplases replases \n"
